include/seqan3/alignment/all.hpp

```diff
@@ -10,20 +10,295 @@
+ * this concept with their respective alphabet(s).
+ *
+ * Two generic scoring schemes are provided:
+ *
+ *   1. seqan3::nucleotide_scoring_scheme that accepts all nucleotides (and any alphabet that is explicitly
+ * convertible to seqan3::dna15)
+ *   2. seqan3::aminoacid_scoring_scheme that accepts all amino acids (and any alphabet that is explicitly convertible
+ * to seqan3::aa27).
+ *
+ * These also support scoring two nucleotides/amino acids of different types and they also support modification of
+ * their scores via `set_()` functions and by returning references to their internal score matrix. You can however
+ * add completely different types, as long as they model seqan3::ScoringScheme.
+ *
+ * The scoring scheme can be configured with the seqan3::align_cfg::scoring element. Since the scoring scheme is
+ * strongly coupled on the sequences to be aligned it can not be defaulted. Thus, it is mandatory for the
+ * the developer to specify the seqan3::align_cfg::scoring configuration.
+ *
+ * <small><i>The base type seqan3::scoring_scheme_base is only implementation detail and not required for most users.
+ * Types that model seqan3::ScoringScheme can (but don't need to!) inherit from it.</i></small>
+ *
+ * ### Scoring gaps
+ *
+ * Gaps are usually not scored on a per-character-basis, because it is widely recognised that the likelihood of
+ * `n` consecutive gaps is much higher than that of `n` individual gaps. This is based on the assumption that
+ * one biological event often introduces more than one gap at a time and that single character gaps are not common
+ * due to other biological factors like frame preservation in protein-coding sequences.
+ *
+ * Therefore, the gaps are not handled by the scoring scheme. SeqAn offers the additional seqan3::gap_scheme
+ * which can be used to set the scores for opening or extending a gap.
+ *
+ * The gap scheme can be configured with the seqan3::align_cfg::gap element. If the configuration is not specified,
+ * the algorithm uses edit distance scores (`-1`) for deletion/insertion.
+ *
+ * ## Computing banded alignments
+ *
+ * SeqAn offers the computation of banded alignments to reduce the running time of the algorithm. This can be
+ * helpful if the region in which the optimal alignment exists is known a priori. To specify the banded alignment
+ * the developer can use the seqan3::align_cfg::band option initialised with a seqan3::static_band.
+ * The seqan3::static_band is constructed with a lower bound and an upper bound. The upper bound must always be greater
+ * than or equal to the lower bound. To position the band, imagine a rectangle where the first sequence is written on
+ * top and the second sequence along the left vertical side. A negative bound implies a start of the band within the
+ * vertical part while a positive bound implies a start within the top part of this rectangle at the respective
+ * position.
```

Comment:
This page could really profit from a few little figures. Can you make a TODO to add these later if you don't have figures lying around for this?

Reply:
later